winlogbeat/eventlog/factory.go

```diff
@@ -48,19 +48,24 @@ func readConfig(
 	validKeys []string,
 ) error {
 	if err := c.Unpack(config); err != nil {
-		return fmt.Errorf("Failed unpacking config. %v", err)
+		return fmt.Errorf("failed unpacking config. %v", err)
+	}
+
+	isValidKey := func(key string) bool {
+		for _, validKey := range validKeys {
+			if key == validKey {
+				return true
+			}
+		}
+		return false
```

Comment:
See libbeat/common/stringset.go: `isValidKey := common.MakeStringSet(validKeys...).Has`

Reply:
👍  I've update this function to accept a StringSet instead.